agent/consul/state/config_entry.go

```diff
@@ -704,6 +710,63 @@ func getReferencedProviderNames(j *structs.IntentionJWTRequirement, s []*structs
 	return providerNames
 }
 
+func validateJWTProviderIsReferenced(tx ReadTxn, kn configentry.KindName, ce structs.ConfigEntry) error {
+	meta := acl.NewEnterpriseMetaWithPartition(
+		kn.EnterpriseMeta.PartitionOrDefault(),
+		acl.DefaultNamespaceName,
+	)
+	entry, ok := ce.(*structs.JWTProviderConfigEntry)
+	if !ok {
+		return fmt.Errorf("invalid jwt provider config entry: %T", entry)
+	}
+
+	_, ixnEntries, err := configEntriesByKindTxn(tx, nil, structs.ServiceIntentions, &meta)
+	if err != nil {
+		return err
+	}
+
+	providerNames, err := collectJWTProviderNames(ixnEntries)
+	if err != nil {
+		return err
+	}
+
+	_, exist := providerNames[entry.Name]
+	if exist {
+		return fmt.Errorf("cannot delete jwt provider config entry referenced by an intention. Provider name:%s", entry.Name)
+	}
+
+	return nil
+}
+
+func collectJWTProviderNames(e []structs.ConfigEntry) (map[string]struct{}, error) {
+	names := make(map[string]struct{})
+
+	for _, entry := range e {
```

Comment:
nit: Conventionally the loop variable name should be a singular or shorter form of the slice. You could do `for _, e := range ee` (go stdlib style) or `for _, e := range entries` but as-is this looks a bit weird.

```suggestion
func collectJWTProviderNames(entries []structs.ConfigEntry) (map[string]struct{}, error) {
	names := make(map[string]struct{})

	for _, entry := range entries {
```